esphome/components/cse7761/cse7761.cpp

```diff
@@ -216,6 +392,10 @@ void CSE7761Component::get_data_() {
     this->voltage_sensor_->publish_state(voltage);
   }
 
+  float freq = (this->data_.frequency) ? ((float) CSE7761_FREF / 8 / this->data_.frequency) : 0;  // Hz
+  if (this->frequency_sensor_ != nullptr) {
```

Comment:
No need to do maths if not going to be used
```suggestion
  if (this->frequency_sensor_ != nullptr) {
    float freq = (this->data_.frequency) ? ((float) CSE7761_FREF / 8 / this->data_.frequency) : 0;  // Hz
```